crates/neon-sys/native/src/neon.cc

```diff
@@ -541,3 +541,24 @@ extern "C" void Neon_EventHandler_Delete(void * thread_safe_cb) {
     neon::EventHandler *cb = static_cast<neon::EventHandler*>(thread_safe_cb);
     cb->close();
 }
+
+extern "C" bool Neon_TryCatch_With(Neon_TryCatchGlue glue_fn, void *cx, void *rust_thunk, v8::Local<v8::Value> *result) {
+  Nan::TryCatch try_catch;
+
+  bool ok;
+  v8::Local<v8::Value> ok_result;
+
+  glue_fn(rust_thunk, cx, &ok, &ok_result);
+
+  if (!try_catch.HasCaught()) {
+    if (ok) {
+      *result = ok_result;
+    } else {
+      *result = Nan::Undefined();
```

Comment:
This should be impossible, right? It would be mean having a `Err(Throw)` without being in a throwing state. If I understand correctly, maybe some equivalent to `unreachable!` instead?

Reply:
I think it's possible because we rely on Neon users cooperatively only using `Err(Throw)` when a throw has actually happened, but AFAIK nothing stops them from lying.

Reply:
Ah, a user could construct `Err(Throw)` manually. C++20 has an `[[unlikely]]` attribute. I'm not sure if that can be used here to any effect.

Reply:
> C++20 has an `[[unlikely]]` attribute. I'm not sure if that can be used here to any effect.

Interesting point, but I'll leave that for a future optimization if we can come up with benchmarks to show it helps. My guess is it doesn't matter because `try_catch` is already probably pretty slow, and it seems like a new enough feature that it might not be fully portable yet across C++ compilers.